app/views/player.js

```diff
@@ -0,0 +1,139 @@
+import Mn from 'backbone.marionette';
+import application from '../application';
+import { timeToString } from '../libs/utils';
+
+const Player = Mn.LayoutView.extend({
+    
+    template: require('./templates/player'),
+
+    ui: {
+        player: 'audio',
+        currentTime: '#current-time',
+        totalTime: '#total-time',
+        progress: '#progress',
+        progressBar: '#progress-bar',
+        volume: '#volume',
+        volumeBar: '#volume-bar',
+        playButton: '#play',
+        trackname: '#trackname'
+    },
+
+    events: {
+        'click #prev': 'prev',
+        'click #play': 'toggle',
+        'click #next': 'next',
+        'click #progress-bar': 'scroll',
+        'click #volume-bar': 'changeVol'
+    },
+
+    onRender: function() {
+        const audio = this.ui.player.get(0);
+        audio.ontimeupdate = this.timeupdate;
+        audio.onended = this.next;
+        audio.onvolumechange = this.onVolumeChange;
+        audio.volume = 0.5;
+        this.listenTo(application.upNext, 'change:currentTrack', function() {
+            const upNext = application.upNext;
+            const currentTrack = upNext.getAttr('currentTrack');
+            if (currentTrack) {
+                this.load(currentTrack);
+            }
+        });
+    },
+
+    load: function(track) {
+        const title = track.get('metas').title;
+        const artist = track.get('metas').artist;
+        let text;
+        if (artist) {
+            text = artist + ' - ' + title;
+        } else {
+            text = title;
+        }
+        this.ui.trackname.text(text);
+        track.getStream(function(url) {
+            application.appLayout.getRegion('player').currentView.play(url);
```

Comment:
Why using this long request? The player view is this view itself, so just call `track.getStream((url) => this.play(url))` should be sufficient (and it break the tight with _appLauyout_).